src/main/java/org/elasticsearch/search/aggregations/reducers/movavg/MovAvgBuilder.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.aggregations.reducers.movavg;
+
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.search.aggregations.reducers.ReducerBuilder;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.elasticsearch.search.aggregations.reducers.BucketHelpers.GapPolicy;
+import static org.elasticsearch.search.aggregations.reducers.movavg.MovAvgModel.Weighting;
+
+public class MovAvgBuilder extends ReducerBuilder<MovAvgBuilder> {
+
+    private String format;
+    private GapPolicy gapPolicy;
+    private Weighting weightingType;
+    private Long window;
+    private Map<String, Object> settings;
+
+    public MovAvgBuilder(String name) {
+        super(name, MovAvgReducer.TYPE.name());
+    }
+
+    public MovAvgBuilder format(String format) {
+        this.format = format;
+        return this;
+    }
+
+    public MovAvgBuilder gapPolicy(GapPolicy gapPolicy) {
+        this.gapPolicy = gapPolicy;
+        return this;
+    }
+
+    public MovAvgBuilder weightingType(Weighting weightingType) {
+        this.weightingType = weightingType;
+        return this;
+    }
+
+    public MovAvgBuilder window(long window) {
+        this.window = window;
+        return this;
+    }
+
+    public MovAvgBuilder settings(Map<String, Object> settings) {
```

Comment:
Good point... I mostly like that version the best.  It could also be used if you don't care about parameters and want the defaults for single/double exp too, like how the match query works. @clintongormley any opinions?